cmk/base/plugins/agent_based/cisco_asa_failover.py

```diff
@@ -0,0 +1,158 @@
+
+#
+# .1.3.6.1.4.1.9.9.147.1.2.1.1.1.2.4  'Failover LAN Interface'
+# .1.3.6.1.4.1.9.9.147.1.2.1.1.1.2.6  'Primary unit (this device)'
+# .1.3.6.1.4.1.9.9.147.1.2.1.1.1.2.7  'Secondary unit'
+# .1.3.6.1.4.1.9.9.147.1.2.1.1.1.3.4  2
+# .1.3.6.1.4.1.9.9.147.1.2.1.1.1.3.6  9     < These two values flip during
+# .1.3.6.1.4.1.9.9.147.1.2.1.1.1.3.7  10    < failover
+# .1.3.6.1.4.1.9.9.147.1.2.1.1.1.4.4  'LAN_FO GigabitEthernet0/0.777'
+# .1.3.6.1.4.1.9.9.147.1.2.1.1.1.4.6  'Active unit'
+# .1.3.6.1.4.1.9.9.147.1.2.1.1.1.4.7  'Standby unit'
+
+
+# [['Failover LAN Interface', '2', 'LAN_FO GigabitEthernet0/0.777'],
+#  ['Primary unit', '9', 'Active unit'],
+#  ['Secondary unit (this device)', '10', 'Standby unit']]
+
+# failover off/ not configured
+# [['Failover LAN Interface', '3', 'not Configured'],
+#  ['Primary unit', '3', 'Failover Off'],
+#  ['Secondary unit (this device)', '3', 'Failover Off']]
+
+from dataclasses import dataclass
+from typing import List, Mapping, Any
+
+from cmk.base.plugins.agent_based.agent_based_api.v1.type_defs import (
+    DiscoveryResult,
+    CheckResult,
+    StringTable,
+)
+
+from cmk.base.plugins.agent_based.agent_based_api.v1 import (
+    register,
+    Service,
+    Result,
+    State,
+    SNMPTree,
+    startswith,
+    contains,
+    any_of,
+)
+
+
+def get_cisco_asa_state_name(st: str) -> str:
+    names = {
+        '1': 'other',
+        '2': 'up',
+        '3': 'down',
+        '4': 'error',
+        '5': 'overTemp',
+        '6': 'busy',
+        '7': 'noMedia',
+        '8': 'backup',
+        '9': 'active',
+        '10': 'standby',
+    }
+    return names.get(st, 'unknown %s' % st)
+
+
+@dataclass
+class Section:
+    failover: bool
+    local_role: str
+    local_status: str
+    local_status_detail: str
+    failver_link_status: str
+    remote_status: str
+
+
+def parse_cisco_asa_failover(string_table: List[StringTable]) -> Section:
+    failover = False
+    for role, status, detail in string_table[0]:
+        if 'this device' in role:
+            if not 'failover off' in detail.lower():
+                failover = True
+            local_role = role.split(' ')[0].lower()
+            local_status = status
+            local_status_detail = detail
+        elif 'failover' in role.lower():
+            failver_link_status = status
+        else:
+            remote_status = status
+
+    return Section(
+        failover=failover,
+        local_role=local_role,
+        local_status=local_status,
+        local_status_detail=local_status_detail,
+        failver_link_status=failver_link_status,
+        remote_status=remote_status
+    )
+
+
+def discovery_cisco_asa_failover(section: Section) -> DiscoveryResult:
+    if section.failover:
+        yield Service()
+
+
+def check_cisco_asa_failover(params: (Mapping[str, Any]), section: Section) -> CheckResult:
+    if isinstance(params, int):  # catch old params
```

Comment:
You can remove this, I would say its better to accept a potential crash here and have easier simpler code instead. Even in Checkmk 1.4, I could not find the mysterious int this is referring to.